ui/components/ui/button/buttons.scss

```diff
@@ -262,7 +262,7 @@ input[type="submit"][disabled] {
     background-color: var(--color-error-default);
 
     &:hover {
-      box-shadow: var(--shadow-size-sm) var(--color-error-shadow);
+      box-shadow: var(--shadow-size-sm) var(--color-shadow-error);
     }
```

Comment:
### After

Shadows still work as expected in deprecated button component


https://github.com/MetaMask/metamask-extension/assets/8112138/2e328890-cd2f-40f7-a46b-f8cc5dd692e8